src/nvim/event/process.c

```diff
@@ -120,14 +120,10 @@ void process_teardown(Loop *loop) FUNC_ATTR_NONNULL_ALL
       // Close handles to process without killing it.
       CREATE_EVENT(loop->events, process_close_handles, 1, proc);
     } else {
-      if (proc->type == kProcessTypeUv) {
-        uv_kill(proc->pid, SIGTERM);
-        proc->term_sent = true;
-        process_stop(proc);
-      } else {  // kProcessTypePty
-        process_close_streams(proc);
-        pty_process_close_master((PtyProcess *)proc);
-      }
+      // Make sure the job kill timer is running and all children are marked
+      // for termination. Otherwise children could block us forever in the
+      // following event loop.
+      process_stop(proc);
```

Comment:
such an approach might work, but it would be better to find out what handle is not properly closed and close them, if possible. (IIRC pty processes are not tracked by uv, but their io streams still are)